### PR TITLE
Adding DSS plugin header for retrieving Nvidia driver & bundle driver in plugin as fallback

### PR DESCRIPTION
**Pull request by vrutz**
https://app.shortcut.com/dataiku/story/214284/eks-requests-2-32-3-is-a-huge-problem-with-github-leading-to-gpu-problem

### CODE REVIEW

File: python-lib/dku_kube/gpu_driver.py [hunk 1]
```diff
@@ -23,10 +23,24 @@ def add_gpu_driver_if_needed(cluster_id, kube_config_path, connection_info, tain
...
 
     # Get the Nvidia driver plugin configuration from the repository
-    nvidia_config_raw = requests.get(
-        "https://raw.githubusercontent.com/NVIDIA/k8s-device-plugin/main/deployments/static/nvidia-device-plugin.yml"
-    ).text
-    nvidia_config = yaml.safe_load(nvidia_config_raw)
+    nvidia_config_response = requests.get(
+        "https://raw.githubusercontent.com/NVIDIA/k8s-device-plugin/main/deployments/static/nvidia-device-plugin.yml",
+        headers={"User-Agent": "DSS EKS Plugin"}
+    )
```
Comment (amandineslx) [inline python-lib/dku_kube/gpu_driver.py:29]: was it intended to remove `.text` here?

File: python-lib/dku_kube/gpu_driver.py [hunk 2]
```diff
@@ -69,3 +83,16 @@ def add_gpu_driver_if_needed(cluster_id, kube_config_path, connection_info, tain
     logging.info("NVIDIA GPU driver config: %s" % yaml.safe_dump(nvidia_config, default_flow_style=False))
 
     run_with_timeout(cmd, env=env, timeout=5)
+
+def get_nvidia_configuration_fallback():
+    logging.warning("Using bundled Nvidia GPU configuration as fallback.")
+    # Check both paths in case the plugin is in development mode
+    nvidia_config_cache_path = os.path.join(os.environ["DKU_CUSTOM_RESOURCE_FOLDER"], "nvidia-device-plugin.yml")
```
Comment (amandineslx) [inline python-lib/dku_kube/gpu_driver.py:90]: how do you know this variable points to the folder `resource` of the plugin?

**Review by amandineslx — COMMENTED** [review]
See questions above

### Tested

* deploy DSS in an elastic fleet with FM
* make new plugin version (commit `33f08f3ab6a5ccb825ad96db4310afda6e6b70b7`)
* update plugin version in DSS
* ✅ start and stop multiple times a cluster with a GPU node pool &rarr; file is correctly pulled
* ✅ the fallback node pool config file is correctly packaged in the plugin

File: python-lib/dku_kube/gpu_driver.py (hunk 1 above)
Comment (vrutz) [reply python-lib/dku_kube/gpu_driver.py:29]: yes, we keep the response object to ensure its status below and then get the text from the response body if it's successful

**Review by amandineslx — APPROVED** [review]
OK for me

**Pull request merged**